docs/advanced-recipes/atom-creators.mdx

```diff
@@ -291,3 +291,61 @@ function EvenCounter() {
   return <>Count was set to an even number {evenCount} times.</>
 }
 ```
+
+## atomWithBroadcast
+
+> `atomWithBroadcast` creates an atom. The atom will be shared between
+> browser tabs and frames, similar to `atomWithStorage` but with the 
+> initialization limitation.
+
+This can be useful when you want the state to interact with each other without
+the use of localStorage and that by just using The Broadcast Channel API allows
+basic communication between browsing contexts (that is, windows, tabs, frames, 
+create a component or iframes) and workers on the same origin.
+
+```ts
+import { atom } from "jotai";
+
+export function atomWithBroadcast<Value>(key: string, initialValue: Value) {
+  const channel = new BroadcastChannel(key);
+
+
+  const baseAtom = atom(initialValue);
+
+  const broadcastAtom = atom<Value, { fromOnMessage: boolean; value: Value }>(
```

Comment:
I guess you don't understand what I mean.

Please try something like this:
```js
const baseAtom = atom(initialValue);
baseAtom.onMount = (setAtom) => {
  ...
});

const broacastAtom = atom(
  (get) => get(baseAtom);
  (get, set, update) => {
      set(baseAtom, update);
      channel.postMessage(get(baseAtom));
  }
);
```

This will support function update too.

Reply:
Oh, okay. I think I missed the fact that `returnedAtom` writes with `fromOnMessage: false`.